packages/graphback/src/buildServices.ts

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import { GraphbackDataProvider, GraphbackCRUDService, CRUDService, GraphbackPubSubModel } from '@graphback/runtime';
+import { PubSub, PubSubEngine } from 'graphql-subscriptions';
+import { ModelDefinition } from '@graphback/core';
+
+export interface ServiceBuilderConfig {
+  defaultDataProvider: GraphbackDataProvider
+  pubSub?: PubSubEngine
+}
+
+export function buildGraphbackServices(options: ServiceBuilderConfig): GraphbackServiceBuilder {
+  const builder = new ServiceBuilder(options)
+
+  return builder
+}
+
+interface ModelDataSourceConfig {
+  [modelName: string]: {
+    service: GraphbackCRUDService
+    provider?: GraphbackDataProvider
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/tslint/config
+class ServiceBuilder implements GraphbackServiceBuilder {
+  private datasourceConfig: ModelDataSourceConfig;
+  private options: ServiceBuilderConfig;
+  public constructor(builderConfig: ServiceBuilderConfig) {
+    this.options = {
+      ...builderConfig,
+      pubSub: new PubSub()
+    }
+
+    const service = new CRUDService(this.options.pubSub)
+    service.setDataProvider(this.options.defaultDataProvider)
+
+    this.datasourceConfig = {
+      default: {
```

Comment:
Sets a default data provider that will be used for each model, unless it is overridden.